[CI] IgnoredSourceFieldMapperTests testStoredNestedSubObjectWithNameOverlappingParentName failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3815 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/cqdk22bf7oyls)
- [elasticsearch-periodic-platform-support #3812 / amazonlinux-2_platform-support-aws](https://gradle-enterprise.elastic.co/s/5heorq6cna63e)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.index.mapper.IgnoredSourceFieldMapperTests.testStoredNestedSubObjectWithNameOverlappingParentName" -Dtests.seed=ADC67C2DF40D32EC -Dtests.locale=es-NI -Dtests.timezone=Africa/Bujumbura -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testStoredNestedSubObjectWithNameOverlappingParentName'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.index.mapper.IgnoredSourceFieldMapperTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: hasPayloads expected:<true> but was:<false>
```

**Issue Reasons:**
- [main] 2 failures in test testStoredNestedSubObjectWithNameOverlappingParentName (1.7% fail rate in 118 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (66.7% fail rate in 3 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

This is the actual failure:


```
java.lang.AssertionError: hasPayloads expected:<false> but was:<true>
	at __randomizedtesting.SeedInfo.seed([B2C4EDF2F8A9577D:E2D1FB3949E3A6A6]:0)
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.apache.lucene.tests.util.LuceneTestCase.assertTermsEquals(LuceneTestCase.java:2133)
	at org.apache.lucene.tests.util.LuceneTestCase.assertTermsEquals(LuceneTestCase.java:2112)
	at org.apache.lucene.tests.util.LuceneTestCase.assertReaderEquals(LuceneTestCase.java:2083)
	at org.elasticsearch.index.mapper.IgnoredSourceFieldMapperTests.validateRoundTripReader(IgnoredSourceFieldMapperTests.java:1235)
	at org.elasticsearch.index.mapper.MapperServiceTestCase.roundTripSyntheticSource(MapperServiceTestCase.java:827)
	at org.elasticsearch.index.mapper.MapperServiceTestCase.syntheticSource(MapperServiceTestCase.java:799)
	at org.elasticsearch.index.mapper.IgnoredSourceFieldMapperTests.testStoredNestedSubObjectWithNameOverlappingParentName(IgnoredSourceFieldMapperTests.java:1219)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
```

For some reason the round trip reader's Terms has payloads for the `path.at.foo` field and the normal reader's terms doesn't have payloads. I don't yet understand how this happens, because at index time the `FieldType` for this field in both indices are the same.

--- a/test/framework/src/main/java/org/elasticsearch/index/mapper/MapperServiceTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/index/mapper/MapperServiceTestCase.java
@@ -788,7 +788,8 @@ public abstract class MapperServiceTestCase extends FieldTypeTestCase {
     }
 
     protected RandomIndexWriter indexWriterForSyntheticSource(Directory directory) throws IOException {
-        return new RandomIndexWriter(random(), directory);
+        // MockAnalyzer (rarely) produces random payloads that lead to failures during assertReaderEquals.
+        return new RandomIndexWriter(random(), directory, new StandardAnalyzer());
     }
 
     protected final String syntheticSource(DocumentMapper mapper, CheckedConsumer<XContentBuilder, IOException> build) throws IOException {

